dask/dataframe/optimize.py

```diff
@@ -20,8 +13,11 @@ def optimize(dsk, keys, **kwargs):
     try:
         from castra import Castra
         dsk3 = fuse_getitem(dsk2, Castra.load_partition, 3)
+        def merge(a, b):
+            return (Castra.load_index, b[1], b[2]) if a[2] == 'index' else a
+        dsk4 = fuse_selections(dsk3, getattr, Castra.load_partition, merge)
```

Comment:
If possible I'd like to keep the `optimize` function as free of clutter as possible.  Can we move this function elsewhere?  

I think I'm still for a more explicit function in dask.dataframe to pull out the index than `getattr`.  If we're going to special case something it makes more sense to me to special case getting out an index than to special case the optimization.